src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/editor.component.ts

```diff
@@ -1670,12 +1670,37 @@ export class EditorComponent extends DataLoadingComponent implements OnDestroy,
       return;
     }
 
-    const thisRange = this.target.segment.range;
-    const thisBounds = this.target.editor.selection.getBounds(thisRange.index);
-
-    const otherRange = this.source.segment.range;
-    const otherBounds = this.source.editor.selection.getBounds(otherRange.index);
-    this.source.editor.scrollingContainer.scrollTop += otherBounds.top - thisBounds.top;
+    const targetRange = this.target.segment.range;
+    const targetSelectionBounds = this.target.editor.selection.getBounds(targetRange.index);
+
+    const sourceRange = this.source.segment.range;
+    const sourceSelectionBounds = this.source.editor.selection.getBounds(sourceRange.index, sourceRange.length);
+
+    const scrollContainer = this.source.editor.scrollingContainer;
+    let newScrollTop: number = scrollContainer.scrollTop + sourceSelectionBounds.top - targetSelectionBounds.top;
+
+    // Check to see if the top of source selection would be visible after the scroll adjustment
+    const sourceTopPosition =
+      scrollContainer.scrollTop -
+      -sourceSelectionBounds.top -
```

Comment:
Should `scrollContainer.scrollTop - -sourceSelectionBounds.top` be `scrollContainer.scrollTop + sourceSelectionBounds.top`?